### ISSUE TITLE
indent: bug in flatTernaryExpressions: true

### ISSUE THREAD

**Issue by aMarCruz** [eslint/eslint#11397]
**Tell us about your environment**
Linux Mint 18.3

* **ESLint Version:** 5.13.0
* **Node Version:** 8.11.2
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** Espree

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,
  rules: {
    'no-undef': 0,
    'no-unused-vars': 0,
    'operator-linebreak': 0,
  }
}
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// test1.js
/* eslint "indent": ["error", 2, {flatTernaryExpressions:true}] */

var a = foo
  ? bar : baz
  ? qux : boop
```

```bash
eslint ./test1.js
```

**What did you expect to happen?**
no errors.
Note the option `flatTernaryExpressions:true` in the config comment.

For this test, I made a simple eslintrc without rules, plugins, etc., but the same thing happens with the typedscript parser and the VS Code ESLint extension.

**What actually happened? Please include the actual, raw output from ESLint.**

/home/xxxxxx/npm/eslint-config-standardize/test1.js
  6:1  error  Expected indentation of 4 spaces but found 2  indent

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

**Are you willing to submit a pull request to fix this bug?**
no.

**Comment by mysticatea** [comment]
Thank you for this issue.

Looks like the intentional behavior.

By design, `flatTernaryExpressions:true` option affects only when a condition and that result are on the same line. I.e.

```js
var a =
  foo ? bar :
  baz ? qux :
  boop
```

The option is to support the style like [Conditional Types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#conditional-types).

```ts
type TypeName<T> =
    T extends string ? "string" :
    T extends number ? "number" :
    T extends boolean ? "boolean" :
    T extends undefined ? "undefined" :
    T extends Function ? "function" :
    "object";
```

**Comment by aMarCruz** [comment]
So there's no way to ESLint succeed with the format of my example? I have problems with the Prettier format and long ternaries.

For example, Prettier will convert this code:

```js
const a = myLongCondition1 ? myLongLongLongValue1
  : myLongCondition2 ? myLongLongLongValue1
  : myLongCondition3
```
on this:

```js
const a = myLongCondition1
  ? myLongLongLongValue1
  : myLongCondition2
  ? myLongLongLongValue1
  : myLongCondition3
```

So, to avoid linting errors, I need to --fix it like this:

```js
const a = myLongCondition1
  ? myLongLongLongValue1
  : myLongCondition2
    ? myLongLongLongValue1
    : myLongCondition3
```

These additional task increases the time of my tests.

Anyway, thanks for you attention.

**Comment by mysticatea** [comment]
If you are using a code formatter, I recommend disabling ESLint rules which format code because the work is duplicated and conflicted. [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier#readme) provides presets to disable format rules which conflict with prettier.

**Comment by aMarCruz** [comment]
@mysticatea , I know.

I'm writing a tool that uses PrettierX as formatter and ESLint as linter, with presets based on StandardJS, but configurable through ESLint. Maybe it becomes a plugin, so that it can be used with the extensions of some editors (mainly VS Code, no extension has decent support for this combination).

Thanks.
